launcher/util.go

```diff
@@ -28,3 +31,20 @@ func FetchImpersonatedToken(ctx context.Context, serviceAccount string, audience
 
 	return []byte(token.AccessToken), nil
 }
+
+func listFilesWithPrefix(targetDir string, prefix string) ([]string, error) {
+	var targetFiles []string
+	err := filepath.WalkDir(targetDir, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
```

Comment:
I guess you also need to do a `d.IsDir` check, otherwise `WalkDir` will skip the path's parent directory, resulting in missing reading potential target files.

Reply:
Yes, Good catch.